### PR TITLE
DOC: GH Action for automatic generation of the docs

### PR DESCRIPTION
**Pull request by arn-all**
Following #28, I set up a github action that builds the docs (after installing the package, so `_matscipy` is importable). Example [here](https://arn-all.github.io/matscipy/). 

It also commits the generated docs to `gh-pages` branch, so it is accessible in html format from within the repo. The repo should be configured to serve the `gh-pages` branch via Github Pages (default behaviour).

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      run: |
+        cd docs
+        make html
+        cd ..
+
+    # Publish built docs to gh-pages branch.
+    # ===============================
+    - name: Commit documentation changes
+      run: |
+        git clone https://github.com/arn-all/matscipy.git --branch gh-pages --single-branch gh-pages
```
Comment (pastewka) [inline .github/workflows/main.yml:39]: This should be libAtoms/matscipy.git

**Comment by pastewka** [comment]
Look good! There is a conflict in `docs/index.rst`. There is also a pointer to `arn-all` that should be `libAtoms`. Can you resolve this? I'll then merge this request.

**Comment by arn-all** [comment]
Sorry ! It should be all good now.

**Comment by pastewka** [comment]
Thanks @arn-all

**Pull request merged**